File: testing/repos/python-example/src/calculator.py
class Calculator:
    """
    A simple calculator class that provides basic arithmetic operations
    """

    def add(self, a: float, b: float) -> float:
        """
        Add two numbers together

        Args:
            a: First number
            b: Second number

        Returns:
            Sum of a and b
        """
        return a + b

    def subtract(self, a: float, b: float) -> float:
        """
        Subtract second number from first number

        Args:
            a: First number
            b: Second number

        Returns:
            Difference between a and b
        """
        return a - b

    def multiply(self, a: float, b: float) -> float:
        """
        Multiply two numbers

        Args:
            a: First number
            b: Second number

        Returns:
            Product of a and b
        """
        return a * b

    def divide(self, a: float, b: float) -> float:
        """
        Divide first number by second number

        Args:
            a: First number
            b: Second number

        Returns:
            Quotient of a divided by b

        Raises:
            ValueError: If b is zero
        """
        if b == 0:
            raise ValueError("Cannot divide by zero")
        return a / b

File: testing/sonar-service-case/repos/go-example/calculator/calculator.go
package calculator

// Calculator represents a basic calculator
type Calculator struct{}

// New creates a new Calculator instance
func New() *Calculator {
	return &Calculator{}
}

// Add returns the sum of two integers
func (c Calculator) Add(a, b int) int {
	return a + b
}

// Subtract returns the difference between two integers
func (c Calculator) Subtract(a, b int) int {
	return a - b
}

// Multiply returns the product of two integers
func (c Calculator) Multiply(a, b int) int {
	return a * b
}

// Divide returns the quotient of two integers
// Panics if divisor is zero
func (c Calculator) Divide(a, b int) int {
	if b == 0 {
		panic("division by zero")
	}
	return a / b
}
